### PR TITLE
Renamed default branch to `main`, update CI

### PR DESCRIPTION
**Pull request by OlegDokuka**
Signed-off-by: Oleh Dokuka <odokuka@vmware.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 [![Download](https://api.bintray.com/packages/spring/jars/io.projectreactor/images/download.svg)](https://bintray.com/spring/jars/io.projectreactor/_latestVersion)
 
-[![Travis CI](https://travis-ci.org/reactor/reactor-core.svg?branch=master)](https://travis-ci.org/reactor/reactor-core)
+[![Travis CI](https://travis-ci.org/reactor/reactor-core.svg?branch=main)](https://travis-ci.org/reactor/reactor-core)
```
Comment (simonbasle) [inline README.md:8]: oh, actually we don't use Travis CI anymore so this badge should be replaced.

Comment (simonbasle) [inline README.md:8]:
```suggestion
[![CI on GHA](https://github.com/reactor/reactor-core/actions/workflows/publish.yml/badge.svg)](https://github.com/reactor/reactor-core/actions/workflows/publish.yml)
```

File: docs/asciidoc/metrics.adoc [hunk 2]
```diff
@@ -37,7 +37,7 @@ Please refer to https://micrometer.io/docs/ref/jvm[Micrometer's documentation] f
 
 Since one scheduler may have multiple executors, every executor metric has a `reactor_scheduler_id` tag.
 
-TIP: Grafana + Prometheus users can use https://raw.githubusercontent.com/reactor/reactor-monitoring-demo/master/dashboards/schedulers.json[a pre-built dashboard] which includes panels for threads, completed tasks, task queues and other handy metrics.
+TIP: Grafana + Prometheus users can use https://raw.githubusercontent.com/reactor/reactor-monitoring-demo/main/dashboards/schedulers.json[a pre-built dashboard] which includes panels for threads, completed tasks, task queues and other handy metrics.
```
Comment (simonbasle) [inline docs/asciidoc/metrics.adoc:40]: this is an unrelated link. not sure about the timeline for these minor / demo projects so for now let's keep `master` here. github will reroute if we do migrate this repo

File: docs/asciidoc/aboutDoc.adoc [hunk 3]
```diff
@@ -44,9 +44,9 @@ essential features) and https://github.com/reactor/reactor-addons/issues[reactor
 (which covers reactor-test and adapters issues).
 
 NOTE: All of Reactor is open source,
-https://github.com/reactor/reactor-core/tree/master/docs/asciidoc[including this
+https://github.com/reactor/reactor-core/tree/main/docs/asciidoc[including this
 documentation]. If you find problems with the docs or if you want to improve them,
-please https://github.com/reactor/.github/blob/master/CONTRIBUTING.md[get involved].
+please https://github.com/reactor/.github/blob/main/CONTRIBUTING.md[get involved].
```
Comment (simonbasle) [inline docs/asciidoc/aboutDoc.adoc:49]: this means we should migrate `.github` repo first, but I'm ok with it.

File: docs/asciidoc/highlight/CHANGES.md [hunk 4]
```diff
@@ -699,7 +699,7 @@ We're going to redesign the old [demo/test suite][demo] into an interactive
 demo web app. If you're confident front-end developer or designer and want to
 help us with it, drop a comment into [the issue][#542] on GitHub.
 
-[test]: https://github.com/isagalaev/highlight.js/tree/master/test
+[test]: https://github.com/isagalaev/highlight.js/tree/main/test
```
Comment (simonbasle) [inline docs/asciidoc/highlight/CHANGES.md:702]: unrelated repo, but `highlith.js` _has_ switched to `main` themselves so I guess this is fine

**Review by simonbasle — CHANGES_REQUESTED** [review]

**Review by simonbasle — CHANGES_REQUESTED** [review]
this needs to be re-targeted to 3.3.x and needs to also change the workflow(s) yaml configurations

File: docs/asciidoc/highlight/CHANGES.md (hunk 4 above)
Comment (OlegDokuka) [reply docs/asciidoc/highlight/CHANGES.md:702]: I checked all the links to ensure they are properly referenced as well

File: docs/asciidoc/metrics.adoc (hunk 2 above)
Comment (simonbasle) [reply docs/asciidoc/metrics.adoc]: please revert that one for now

File: docs/asciidoc/processors.adoc [hunk 5]
```diff
@@ -77,7 +77,8 @@ draining it.
 TIP: One way of publishing events onto different threads is to use the `EmitterProcessor`
 combined with `publishOn(Scheduler)`. This can for example replace the former `TopicProcessor`,
 which was using `Unsafe` operations and has been moved to
-https://github.com/reactor/reactor-addons/tree/master/reactor-extra/src/main/java/reactor/extra/processor[reactor-extra]
+https://github.com/reactor/reactor-addons/tree/main/reactor-extra/src/main/java/reactor
```
Comment (simonbasle) [inline docs/asciidoc/processors.adoc:80]: ok, we should make the switch almost at the same time for addons and core

File: docs/asciidoc/aboutDoc.adoc (hunk 3 above)
Comment (simonbasle) [reply docs/asciidoc/aboutDoc.adoc:49]: (this has been done)

**Review by simonbasle — APPROVED** [review]
looks good to me, approving and I'll coordinate the merging with the actual act of renaming, probably in a few days (let's rename in lower-traffic projects first)

**Comment by simonbasle** [comment]
merging checklist:
 - [ ] perform the actual renaming
 - [ ] update `snapshots` and `releases` environments branch restriction
 - [ ] merge this PR
 - [ ] forward merge into 3.4.5
